### ISSUE TITLE
Duplicate pos file path  in fluent.conf with mounted-file

### ISSUE THREAD

**Issue by cw-sakamoto** [vmware/kube-fluentd-operator#29]
I tried with the settings below, the fluentd container became crashLoopback.

Calcurating hash in https://github.com/vmware/kube-fluentd-operator/blob/master/config-reloader/processors/mounted_file.go#L110 seems to not use container id.

fluentd log
```
2018-08-27 06:58:06 +0000 [info]: parsing config file is succeeded path="/fluentd/etc/fluent.conf"
2018-08-27 06:58:06 +0000 [info]: adding rewrite_tag_filter rule: unit [#<Fluent::PluginHelper::RecordAccessor::Accessor:0x007f8d06dd2878 @keys="unit">, /^(.+)$/, "", "systemd.$1"]
2018-08-27 06:58:06 +0000 [info]: adding rewrite_tag_filter rule: kubernetes_namespace_container_name [#<Fluent::PluginHelper::RecordAccessor::Accessor:0x007f8d023005e8 @keys="kubernetes_namespace_container_name">, /^(.+)$/, "", "kube.$1"]
2018-08-27 06:58:07 +0000 [warn]: both of Plugin @id and path for <storage> are not specified. Using on-memory store.
2018-08-27 06:58:07 +0000 [error]: config error file="/fluentd/etc/fluent.conf" error_class=Fluent::ConfigError error="Other 'in_tail' plugin already use same pos_file path: plugin_id = object:3fc680ff6dfc, pos_file path = /var/log/kfotail-e2ed04fd00276fdd0b2583231b9044c0c12c45b2.pos"
```

Deployment:
```
apiVersion: apps/v1beta2 # for versions before 1.8.0 use apps/v1beta1
kind: Deployment
metadata:
  name: hello-logger-deployment
  labels:
    app: hello-logger
spec:
  replicas: 2
  selector:
    matchLabels:
      app: hello-logger
  template:
    metadata:
      labels:
        app: hello-logger
        msg: hello
    spec:
      containers:
      - image: ubuntu
        name: greeter-hoge
        command: [ "bash", "-c"]
        args:
        - |
          output1=/var/log/hello1.log
          output2=/var/log/hello2.log
          while true; do
            echo `date -R` [INFO] "Random hello1 number $((var1++)) to file" > ${output1}
            echo `date -R` [INFO] "Random hello2 number $((var2++)) to file" > ${output2}
            sleep 2;
            [[ $(($var1 % 100)) == 0 ]] && :> ${output1} ;
            [[ $(($var2 % 100)) == 0 ]] && :> ${output2} ;
          done
        volumeMounts:
        - mountPath: /var/log
          name: logs
      - image: ubuntu
        name: greeter-foo
        command: [ "bash", "-c"]
        args:
        - |
          output3=/var/log/hello3.log
          while true; do
            echo `date -R` [INFO] "Random hello3 number $((var3++)) to file" > ${output3}
            sleep 2;
            [[ $(($var3 % 100)) == 0 ]] && :> ${output3} ;
          done
        volumeMounts:
        - mountPath: /var/log
          name: logs
      volumes:
      - name: logs
        emptyDir: {}
```

configmap for fluent.conf
```
<source>
  @type mounted-file
  path /var/log/hello1.log
  labels msg=hello
</source>
 
<match **>
  @type null
</match>
```

/fluentd/etc/fluent.conf in reloader(fluentd)
```
# dont modify this file when building on top of the image

<system>
  log_level info

  # needed to enable /api/config.reload
  rpc_endpoint 127.0.0.1:24444
</system>

# you can turn this on for debug
# <match fluent.**>
#   @type stdout
# </match>


# OS-level services
@include systemd.conf

# docker container logs
@include kubernetes.conf

# enrich docker logs with k8s metadata
@include kubernetes-postprocess.conf


#################
# Namespace pre-processing
#################
<source>
  @type tail
  path /var/lib/kubelet/pods/dde1207b-a9bf-11e8-a89a-0a391dd99bb8/volumes/kubernetes.io~empty-dir/logs/hello1.log
  pos_file /var/log/kfotail-e2ed04fd00276fdd0b2583231b9044c0c12c45b2.pos
  read_from_head true
  tag kube.test.hello-logger-deployment-7c5c7759fb-9nh9n.greeter-hoge

  <parse>
    @type none
  </parse>
</source>

<filter kube.test.hello-logger-deployment-7c5c7759fb-9nh9n.greeter-hoge>
  @type record_modifier
  remove_keys dummy_

  <record>
    dummy_ ${record['stream']='/var/log/hello1.log'; record['kubernetes']={'container_name'=>'greeter-hoge','host'=>'ip-10-0-4-93.ap-northeast-1.compute.internal','namespace_name'=>'test','pod_id'=>'dde1207b-a9bf-11e8-a89a-0a391dd99bb8','pod_name'=>'hello-logger-deployment-7c5c7759fb-9nh9n'}; record['container_info']='deac11d9b446470dd58bf76521a9196531b4d4b1'; record['kubernetes']['labels']={'app'=>'hello-logger','msg'=>'hello','pod-template-hash'=>'3717331596'}; record['kubernetes']['namespace_labels']={}}
  </record>
</filter>

<source>
  @type tail
  path /var/lib/kubelet/pods/dde1207b-a9bf-11e8-a89a-0a391dd99bb8/volumes/kubernetes.io~empty-dir/logs/hello1.log
  pos_file /var/log/kfotail-e2ed04fd00276fdd0b2583231b9044c0c12c45b2.pos
  read_from_head true
  tag kube.test.hello-logger-deployment-7c5c7759fb-9nh9n.greeter-foo

  <parse>
    @type none
  </parse>
</source>

<filter kube.test.hello-logger-deployment-7c5c7759fb-9nh9n.greeter-foo>
  @type record_modifier
  remove_keys dummy_

  <record>
    dummy_ ${record['stream']='/var/log/hello1.log'; record['kubernetes']={'container_name'=>'greeter-foo','host'=>'ip-10-0-4-93.ap-northeast-1.compute.internal','namespace_name'=>'test','pod_id'=>'dde1207b-a9bf-11e8-a89a-0a391dd99bb8','pod_name'=>'hello-logger-deployment-7c5c7759fb-9nh9n'}; record['container_info']='deac11d9b446470dd58bf76521a9196531b4d4b1'; record['kubernetes']['labels']={'app'=>'hello-logger','msg'=>'hello','pod-template-hash'=>'3717331596'}; record['kubernetes']['namespace_labels']={}}
  </record>
</filter>

<source>
  @type tail
  path /var/lib/kubelet/pods/e208e61a-a9bf-11e8-a89a-0a391dd99bb8/volumes/kubernetes.io~empty-dir/logs/hello1.log
  pos_file /var/log/kfotail-f2edf6c3f456e55f2a38a79fc1fa21e351e52603.pos
  read_from_head true
  tag kube.test.hello-logger-deployment-7c5c7759fb-dqngq.greeter-hoge

  <parse>
    @type none
  </parse>
</source>

<filter kube.test.hello-logger-deployment-7c5c7759fb-dqngq.greeter-hoge>
  @type record_modifier
  remove_keys dummy_

  <record>
    dummy_ ${record['stream']='/var/log/hello1.log'; record['kubernetes']={'container_name'=>'greeter-hoge','host'=>'ip-10-0-4-93.ap-northeast-1.compute.internal','namespace_name'=>'test','pod_id'=>'e208e61a-a9bf-11e8-a89a-0a391dd99bb8','pod_name'=>'hello-logger-deployment-7c5c7759fb-dqngq'}; record['container_info']='aa33c2ab3e30423fbfe283fc7a11af676d1aac8a'; record['kubernetes']['labels']={'app'=>'hello-logger','msg'=>'hello','pod-template-hash'=>'3717331596'}; record['kubernetes']['namespace_labels']={}}
  </record>
</filter>

<source>
  @type tail
  path /var/lib/kubelet/pods/e208e61a-a9bf-11e8-a89a-0a391dd99bb8/volumes/kubernetes.io~empty-dir/logs/hello1.log
  pos_file /var/log/kfotail-f2edf6c3f456e55f2a38a79fc1fa21e351e52603.pos
  read_from_head true
  tag kube.test.hello-logger-deployment-7c5c7759fb-dqngq.greeter-foo

  <parse>
    @type none
  </parse>
</source>

<filter kube.test.hello-logger-deployment-7c5c7759fb-dqngq.greeter-foo>
  @type record_modifier
  remove_keys dummy_

  <record>
    dummy_ ${record['stream']='/var/log/hello1.log'; record['kubernetes']={'container_name'=>'greeter-foo','host'=>'ip-10-0-4-93.ap-northeast-1.compute.internal','namespace_name'=>'test','pod_id'=>'e208e61a-a9bf-11e8-a89a-0a391dd99bb8','pod_name'=>'hello-logger-deployment-7c5c7759fb-dqngq'}; record['container_info']='aa33c2ab3e30423fbfe283fc7a11af676d1aac8a'; record['kubernetes']['labels']={'app'=>'hello-logger','msg'=>'hello','pod-template-hash'=>'3717331596'}; record['kubernetes']['namespace_labels']={}}
  </record>
</filter>




#################
# Generated based on annotated kube-system namespace
#################
@include kube-system.conf
#################


#################
# Generated based on namespace annotations
#################
@include ns-test.conf

#################


<match **>
  # prevent fluentd from reporting every unmatched tag
  @type null
</match>
```

**Comment by jvassev** [comment]
Hi @cw-sakamoto,

Thank you for the detailed bug report. You've also located the issue. I've provided the fix in d96a7d6  - now the hash is built using the pod + container name.
You can try out this image tag: `v1.6.0-17-gd96a7d6`

**Comment by cw-sakamoto** [comment]
Thanks! I confirm fixed.
